### PR TITLE
OF-2559: Use Netty for LocalIncomingServerSessionTest

### PR DESCRIPTION
**Pull request by guusdk**
Prior to this commit, the unit test for LocalIncomingServerSession used the old, pre-Netty code. With this commit, the Netty code is being used instead.

Insteead of the old blocking code, the test now uses a (Netty-backed) ConnectionListener directly. To identify the LocalIncomingServderSession instance that is to be created, a streamID value is used, which is collected by the dummy server peer implementation for that purpose.

### CODE REVIEW

**Comment by guusdk** [comment]
I'm slighly concerned by the removal of the now-unused `ConnectionConfigurationAnswer` method in the test fixture. Shouldn't that be needed?

For me locally, all these tests are green when this PR is combined with the changes from #2229, #2230 and #2231.

File: xmppserver/src/test/java/org/jivesoftware/openfire/session/LocalIncomingServerSessionTest.java
Comment (AlexGidman) [inline xmppserver/src/test/java/org/jivesoftware/openfire/session/LocalIncomingServerSessionTest.java:38]: Unused imports:

`import org.mockito.stubbing.Answer;`
`import org.mockito.invocation.InvocationOnMock;`
`import org.jivesoftware.openfire.spi.ConnectionConfiguration;`

**Review by AlexGidman — COMMENTED** [review]
Changes make sense: tests not passing without other changes mentioned so should probably wait until they have also been merged.

Can we delete SocketAcceptThread, BlockingAcceptingMode, SocketAcceptingMode, and ServerSocketReader now? It looks like these are all no longer used.

Some elements of the former implementation (SocketConnection etc) are still used for ServerDialback and other features of Openfire (perhaps these can be ticketed in OF Jira?)

File: xmppserver/src/test/java/org/jivesoftware/openfire/session/LocalIncomingServerSessionTest.java
Comment (viv) [inline xmppserver/src/test/java/org/jivesoftware/openfire/session/LocalIncomingServerSessionTest.java:111]: I looked into your concern about removing `ConnectionConfigurationAnswer`. What I've seen is that `ConnectionConfigurationAnswer` was used to mock the call to `generateConnectionConfiguration()` on the `ConnectionListener`. 

With your new commit you are supplying the configuration previously defined in the `ConnectionConfigurationAnswer` directly to `ConnectionListener`, when `generateConnectionConfiguration()` is called it uses the config you passed in:

```
connectionListener = new ConnectionListener(ConnectionType.SOCKET_S2S,
                ConnectionSettings.Server.OLD_SSLPORT,
                ConnectionManagerImpl.DEFAULT_SERVER_SSL_PORT,
                ConnectionSettings.Server.ENABLE_OLD_SSLPORT,
                "xmpp.server.processing.threads",
                null,
                JiveGlobals.getProperty(ConnectionSettings.Server.TLS_POLICY),
                ConnectionSettings.Server.AUTH_PER_CLIENTCERT_POLICY,
                null,
                identityStore.getConfiguration(),
                trustStore.getConfiguration(),
                ConnectionSettings.Server.COMPRESSION_SETTINGS);
```

Not mocking the `ConnectionListener` makes sense to me in the context of an incoming server test.

**Review by viv — APPROVED** [review]

File: xmppserver/src/test/java/org/jivesoftware/openfire/session/LocalIncomingServerSessionTest.java
Comment (guusdk) [reply xmppserver/src/test/java/org/jivesoftware/openfire/session/LocalIncomingServerSessionTest.java:111]: Yeah, you're probably right. I was mostly worried about 'loosing' the ability to configure TLS-specifics - but I guess we can do that to the properties (as is done with most other configuration in this same test).

**Comment by guusdk** [comment]
I very much want the old networking code to be removed, but didn't want to do it as part of this PR. As per @AlexGidman's suggestion, I have raised a new JIRA ticket for the removal effort: https://igniterealtime.atlassian.net/browse/OF-2635

The review suggestions were applied, and this PR has been rebased on the latest main.

**Comment by guusdk** [comment]
My optimizations seem to have improved the LocalIncomingServerSessionTest (unexpectedly), but have also made the LocalOutgoingServerSessionTest fail its 13'th test - but only if it is run as part of multiple tests. I'm guessing that I've somehow made tests interdependant (which is bad).

**Comment by guusdk** [comment]
Ah, I suspect that the new tests were failing off and on again from a race condition, not because session state was carried over. The race condition is fixed in commit 8f2f4f2

**Comment by guusdk** [comment]
This PR is running into issues that are likely addressed in another PR: #2226. That PR is running into issues fixed by this PR.

To resolve this chicken/egg problem, I'm going to merge this PR, and rebase the other PR.

**Pull request merged**
